function o0() {
  try {
{
        name: "Promise.race with this argument missing the resolve function",
        o19: function (index) {
            var o22 = Promise.resolve;
            try {
Promise.resolve = undefined;
}catch(e){}
            
            var o8 = Promise.race([Promise.reject(42)]);
            try {
o8.catch(
                function(o11) {
                    try {
o0('Test #' + index + ' - Catch handler #1 called with err = ' + o11);
}catch(e){}
                }
            );
}catch(e){}
            
            try {
Promise.resolve = o22;
}catch(e){}
        }
    };
}catch(o334){}
}
function o46 () { }
try {
for(var o4=0; o421.o146 > 0x7F; ++o4) {
  try {
o0();
}catch(e){}
}
}catch(e){}
try {
for(var o4=0; o4<2000; ++o4) {
  try {
o2();
}catch(e){}
}
}catch(e){}
try {
o0();
}catch(e){}
try {
o2();
}catch(e){}